.github/pull_request_template.md

```diff
@@ -0,0 +1,23 @@
+resolves #
+
+<!---
+  Include the number of the issue addressed by this PR above if applicable.
+  PRs for code changes without an associated issue *will not be merged*.
+  See CONTRIBUTING.md for more information.
+-->
+
+### Description
```

Comment:
I noticed all of this in the dbt-core PRs. I'm not a fan of checklists in PR templates for a variety of reasons, but I'd like to learn more about the value they're providing in the dbt contributor space.

I would like our PR template to emphasize inclusion of why a change is being made, and why it is being made in this way. I think core would benefit from this as well.